src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -25,10 +25,9 @@
 CLI_PACKAGE_NAME = 'azure-cli'
 COMPONENT_PREFIX = 'azure-cli-'
 
-SSLERROR_TEMPLATE = ('Certificate verification failed. This typically happens when using Azure CLI behind a proxy '
-                     'that intercepts traffic with a self-signed certificate. '
-                     # pylint: disable=line-too-long
-                     'Please add this certificate to the trusted CA bundle. More info: https://docs.microsoft.com/cli/azure/use-cli-effectively#work-behind-a-proxy.')
+SSLERROR_TEMPLATE = ('An SSL error occurred. This typically happens when using Azure CLI behind a proxy or firewall. '
+                     'For more information, see: '
+                     'https://learn.microsoft.com/cli/azure/use-cli-effectively#work-behind-a-proxy')
```

Comment:
Can we distinguish the firewall error and certificate error further based on error message?

The certificate error is `Caused by SSLError(SSLCertVerificationError`.
The firewall error is `Caused by SSLError(SSLEOFError`. In this case, we can suggest users adding endpoints to allowlist: https://learn.microsoft.com/en-us/cli/azure/azure-cli-endpoints?tabs=azure-cloud

Reply:
Firewall is a special type of proxy. I think we can elaborate it in the doc, but keep the error message simple.